modules/web/src/app/node-data/basic/provider/anexia/component.ts

```diff
@@ -208,14 +275,36 @@ export class AnexiaBasicNodeDataComponent extends BaseFormValidator implements O
 
   private _setDefaultTemplate(templates: AnexiaTemplate[]): void {
     this.isLoadingTemplates = false;
-    this.templates = _.sortBy(templates, t => t.id.toLowerCase()).map(t => t.id);
-    let selectedTemplate = this._nodeDataService.nodeData.spec.cloud.anexia.templateID;
+    this.templateIDs = _.sortBy(templates, t => t.id.toLowerCase()).map(t => t.id);
+    this.templateNames = _.sortBy(templates, t => t.name.toLowerCase()).map(t => t.name);
+    let selectedTemplate =
+      this._nodeDataService.nodeData.spec.cloud.anexia.template ||
+      this._nodeDataService.nodeData.spec.cloud.anexia.templateID;
+    if (!selectedTemplate && this.templateNames.length > 0) {
+      selectedTemplate = this.templateNames[0];
+    }
+    if (selectedTemplate) {
+      this.form.get(Controls.Template).setValue({main: selectedTemplate});
+    }
+    this._cdr.detectChanges();
+  }
 
-    if (!selectedTemplate && this.templates.length > 0) {
-      selectedTemplate = this.templates[0];
+  private _setDefaultDiskType(diskTypes: AnexiaDiskType[]): void {
+    this.isLoadingDiskTypes = false;
+    this.diskTypes = diskTypes.map(dt => dt.id);
+
+    for (let i = 0; i < this.disks.length; i++) {
+      if (
+        this._nodeDataService.nodeData.spec.cloud.anexia.disks.length > i &&
+        this._nodeDataService.nodeData.spec.cloud.anexia.disks[i].performanceType
+      ) {
+        this.disks
+          .at(i)
+          .get('performanceType')
+          .setValue({main: this._nodeDataService.nodeData.spec.cloud.anexia.disks[i].performanceType});
+      }
```

Comment:
I think we don't need this here. `performanceType` is already set for `disks` inside the `_init` method. Or am I missing something?